### PR TITLE
Add cleanup for revoked tokens with appropriate indexes

### PR DESCRIPTION
**Pull request by franjo-zadelj-trikoder**
Cleanup for revoked tokens + indexes

### CODE REVIEW

File: Manager/AccessTokenManagerInterface.php [hunk 1]
```diff
@@ -6,6 +6,9 @@
 
 use Trikoder\Bundle\OAuth2Bundle\Model\AccessToken;
 
+/**
+ * @method int clearRevoked()
```
Comment (X-Coder264) [inline Manager/AccessTokenManagerInterface.php:10]: @franjo-zadelj-trikoder Add a note explaining that not defining this method is deprecated since version 3.2 and that the method will be added to the interface in v4.

Example from Symfony:

https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/HttpKernel/KernelInterface.php#L25

Comment (X-Coder264) [reply Manager/AccessTokenManagerInterface.php:10]: and refresh and auth code manager interfaces should receive the same docblock. Currently you've put this only on the access token manager interface.

File: Manager/InMemory/AuthorizationCodeManager.php [hunk 2]
```diff
@@ -36,4 +36,15 @@ public function clearExpired(): int
 
         return $count - \count($this->authorizationCodes);
     }
+
+    public function clearRevoked(): int
+    {
+        $count = \count($this->authorizationCodes);
+
+        $this->accessTokens = array_filter($this->authorizationCodes, static function (AuthorizationCode $accessToken): bool {
```
Comment (X-Coder264) [inline Manager/InMemory/AuthorizationCodeManager.php:44]: @franjo-zadelj-trikoder `$this->accessTokens` and `AuthorizationCode $accessToken` ?

File: Manager/InMemory/AuthorizationCodeManager.php [hunk 3]
```diff
@@ -36,4 +36,15 @@ public function clearExpired(): int
...
+            return !$accessToken->isRevoked();
+        });
+
+        return $count - \count($this->accessTokens);
```
Comment (X-Coder264) [inline Manager/InMemory/AuthorizationCodeManager.php:48]: @franjo-zadelj-trikoder `$this->accessTokens`?

please add tests for all in memory implementations.

all three implementations already have unit tests:


https://github.com/trikoder/oauth2-bundle/tree/v3.x/Tests/Unit

File: Manager/InMemory/RefreshTokenManager.php [hunk 4]
```diff
@@ -42,4 +42,15 @@ public function clearExpired(): int
 
         return $count - \count($this->refreshTokens);
     }
+
+    public function clearRevoked(): int
+    {
+        $count = \count($this->refreshTokens);
+
+        $this->accessTokens = array_filter($this->refreshTokens, static function (RefreshToken $accessToken): bool {
```
Comment (X-Coder264) [inline Manager/InMemory/RefreshTokenManager.php:50]: @franjo-zadelj-trikoder `$this->accessTokens` and `RefreshToken $accessToken`?

File: Manager/InMemory/RefreshTokenManager.php [hunk 5]
```diff
@@ -42,4 +42,15 @@ public function clearExpired(): int
...
+            return !$accessToken->isRevoked();
+        });
+
+        return $count - \count($this->accessTokens);
```
Comment (X-Coder264) [inline Manager/InMemory/RefreshTokenManager.php:54]: @franjo-zadelj-trikoder `$this->accessTokens`?

File: Resources/config/doctrine/model/AccessToken.orm.xml [hunk 6]
```diff
@@ -14,6 +14,11 @@
         <field name="userIdentifier" type="string" length="128" nullable="true" />
         <field name="scopes" type="oauth2_scope" nullable="true" />
         <field name="revoked" type="boolean" />
+        <indexes>
+            <index name="idx_oauth2_access_token_client" columns="client"/>
```
Comment (X-Coder264) [inline Resources/config/doctrine/model/AccessToken.orm.xml:18]: @franjo-zadelj-trikoder this is redundant, doctrine relationships already get an index by default

File: Resources/config/doctrine/model/AuthorizationCode.orm.xml [hunk 7]
```diff
@@ -14,6 +14,11 @@
         <field name="userIdentifier" type="string" length="128" nullable="true" />
         <field name="scopes" type="oauth2_scope" nullable="true" />
         <field name="revoked" type="boolean" />
+        <indexes>
+            <index name="idx_oauth2_authorization_code_client" columns="client"/>
```
Comment (X-Coder264) [inline Resources/config/doctrine/model/AuthorizationCode.orm.xml:18]: @franjo-zadelj-trikoder this is redundant, doctrine relationships already get an index by default

File: Resources/config/doctrine/model/RefreshToken.orm.xml [hunk 8]
```diff
@@ -12,6 +12,11 @@
         </id>
         <field name="expiry" type="datetime_immutable" />
         <field name="revoked" type="boolean" />
+        <indexes>
+            <index name="idx_oauth2_refresh_token_access_token" columns="access_token"/>
```
Comment (X-Coder264) [inline Resources/config/doctrine/model/RefreshToken.orm.xml:16]: @franjo-zadelj-trikoder this is redundant, doctrine relationships already get an index by default

File: Resources/config/doctrine/model/AccessToken.orm.xml [hunk 9]
```diff
@@ -14,6 +14,11 @@
...
+            <index name="idx_oauth2_access_token_userIdentifier" columns="userIdentifier"/>
```
Comment (X-Coder264) [inline Resources/config/doctrine/model/AccessToken.orm.xml:19]: @HypeMC do you remember why we've reverted this index in the past? There were some issues with Doctrine index naming IIRC.

File: Tests/Acceptance/ClearRevokedTokensCommandTest.php [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+use Trikoder\Bundle\OAuth2Bundle\Model\RefreshToken;
+use Trikoder\Bundle\OAuth2Bundle\Tests\Fixtures\FixtureFactory;
+
+final class ClearRevokedTokensCommandTest extends AbstractAcceptanceTest
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        timecop_freeze(new \DateTimeImmutable());
```
Comment (X-Coder264) [inline Tests/Acceptance/ClearRevokedTokensCommandTest.php:25]: @franjo-zadelj-trikoder why is the time freezed for this test?

File: Command/ClearRevokedTokensCommand.php [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+        $clearExpiredAuthCodes = $input->getOption('auth-codes');
+
+        if (!$clearExpiredAccessTokens && !$clearExpiredRefreshTokens && !$clearExpiredAuthCodes) {
+            $clearExpiredAccessTokens = true;
+            $clearExpiredRefreshTokens = true;
+            $clearExpiredAuthCodes = true;
+        }
+
+        if (true === $clearExpiredAccessTokens) {
+            $affected = $this->accessTokenManager->clearRevoked();
```
Comment (X-Coder264) [inline Command/ClearRevokedTokensCommand.php:84]: @franjo-zadelj-trikoder there must be a `method_exists` check here and if the method does not exist on the implementation we should warn the user about it, instead of just blowing up. This applies to the refresh and auth code revokation logic below too.

File: Manager/Doctrine/RefreshTokenManager.php [hunk 12]
```diff
@@ -47,4 +47,13 @@ public function clearExpired(): int
             ->getQuery()
             ->execute();
     }
+
+    public function clearRevoked(): int
+    {
+        return $this->entityManager->createQueryBuilder()
+            ->delete(RefreshToken::class, 'at')
```
Comment (X-Coder264) [inline Manager/Doctrine/RefreshTokenManager.php:54]: @franjo-zadelj-trikoder `rt`, not `at`.

File: Manager/Doctrine/AuthorizationCodeManager.php [hunk 13]
```diff
@@ -47,4 +47,13 @@ public function clearExpired(): int
             ->getQuery()
             ->execute();
     }
+
+    public function clearRevoked(): int
+    {
+        return $this->entityManager->createQueryBuilder()
+            ->delete(AuthorizationCode::class, 'at')
```
Comment (X-Coder264) [inline Manager/Doctrine/AuthorizationCodeManager.php:54]: @franjo-zadelj-trikoder `ac,` not `at`.

**Review by X-Coder264 — CHANGES_REQUESTED** [review]

File: Resources/config/doctrine/model/AccessToken.orm.xml (hunk 9 above)
Comment (HypeMC) [reply Resources/config/doctrine/model/AccessToken.orm.xml:19]: @X-Coder264 Yep, Doctrine didn't take into account the naming strategy https://github.com/trikoder/oauth2-bundle/pull/197 , this needs to be removed for now.

File: Tests/Acceptance/ClearRevokedTokensCommandTest.php [hunk 14]
```diff
@@ -0,0 +1,193 @@
...
+        );
+        $this->assertInstanceOf(
+            AuthorizationCode::class,
+            $this->client->getContainer()->get(AuthorizationCodeManagerInterface::class)->find(
+                FixtureFactory::FIXTURE_AUTH_CODE_REVOKED
+            )
+        );
+    }
+
+    public function testClearExpiredRefreshTokens(): void
```
Comment (HypeMC) [inline Tests/Acceptance/ClearRevokedTokensCommandTest.php:113]: @franjo-zadelj-trikoder 
```suggestion
    public function testClearRevokedRefreshTokens(): void
```

File: Tests/Acceptance/ClearRevokedTokensCommandTest.php [hunk 15]
```diff
@@ -0,0 +1,193 @@
...
+        );
+        $this->assertInstanceOf(
+            AuthorizationCode::class,
+            $this->client->getContainer()->get(AuthorizationCodeManagerInterface::class)->find(
+                FixtureFactory::FIXTURE_AUTH_CODE_REVOKED
+            )
+        );
+    }
+
+    public function testClearExpiredAuthCodes(): void
```
Comment (HypeMC) [inline Tests/Acceptance/ClearRevokedTokensCommandTest.php:142]: @franjo-zadelj-trikoder 
```suggestion
    public function testClearRevokedAuthCodes(): void
```

File: Manager/Doctrine/AccessTokenManager.php [hunk 16]
```diff
@@ -47,4 +47,13 @@ public function clearExpired(): int
             ->getQuery()
             ->execute();
     }
+
+    public function clearRevoked(): int
+    {
+        return $this->entityManager->createQueryBuilder()
+            ->delete(AccessToken::class, 'at')
+            ->where('at.revoked = 1')
```
Comment (HypeMC) [inline Manager/Doctrine/AccessTokenManager.php:55]: @franjo-zadelj-trikoder DQL supports [boolean literals](https://www.doctrine-project.org/projects/doctrine-orm/en/2.7/reference/dql-doctrine-query-language.html#literal-values):
```suggestion
            ->where('at.revoked = true')
```

This makes the code more consistent and easier to read IMO, @X-Coder264 your thoughts?

**Review by HypeMC — CHANGES_REQUESTED** [review]

File: Resources/config/doctrine/model/AccessToken.orm.xml (hunk 9 above)
Comment (franjo-zadelj-trikoder) [reply Resources/config/doctrine/model/AccessToken.orm.xml:19]: @X-Coder264 @HypeMC I don't understand what's the problem with this? The only bug I found is you can't have more indexes with the same name, even on different tables.

File: Tests/Acceptance/ClearRevokedTokensCommandTest.php (hunk 10 above)
Comment (franjo-zadelj-trikoder) [reply Tests/Acceptance/ClearRevokedTokensCommandTest.php:25]: @X-Coder264 why is it frozen here vendor/trikoder/oauth2-bundle/Tests/Acceptance/ClearExpiredTokensCommandTest.php

File: Resources/config/doctrine/model/AccessToken.orm.xml (hunk 9 above)
Comment (X-Coder264) [reply Resources/config/doctrine/model/AccessToken.orm.xml:19]: @franjo-zadelj-trikoder https://github.com/trikoder/oauth2-bundle/issues/196

Comment (HypeMC) [reply Resources/config/doctrine/model/AccessToken.orm.xml:19]: @franjo-zadelj-trikoder Please keep in mind that this is an open source bundle used by a lot of projects, not just our internal ones. Though everything might work in our case, we can't expect that everyone uses the same naming strategy as we do. To ensure compatibility with different naming strategies on different projects we must not add this for now. Once the bug gets fixed, we can consider adding it back.

**Review by HypeMC — CHANGES_REQUESTED** [review]

File: Tests/Acceptance/ClearRevokedTokensCommandTest.php (hunk 10 above)
Comment (X-Coder264) [reply Tests/Acceptance/ClearRevokedTokensCommandTest.php:25]: @franjo-zadelj-trikoder Because whether the token is expired or not depends on the time, which is not the case with the revoke functionality.

File: Command/ClearRevokedTokensCommand.php [hunk 17]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    private function checkMethod(OutputInterface $output, object $obj, string $methodName = 'clearRevoked'): bool
+    {
+        $exists = method_exists($obj, $methodName);
+
+        if (!$exists) {
+            $output->writeln(
+                sprintf(
+                    '<comment>Method "%s:%s()" will be required in the next release. Skipping for now...</comment>',
```
Comment (X-Coder264) [inline Command/ClearRevokedTokensCommand.php:123]: @franjo-zadelj-trikoder Not in the next release - in the next *major* release.

File: Tests/Acceptance/DoctrineAccessTokenManagerTest.php [hunk 18]
```diff
@@ -109,45 +232,79 @@ public function testClearExpiredWithRefreshToken(): void
...
-        ];
+        $client = new Client('client', 'secret');
+        $em->persist($client);
+        $em->flush();
 
-        return [
-            'input' => array_merge($validRefreshTokens, $expiredRefreshTokens),
-            'output' => $expiredRefreshTokens,
-        ];
+        timecop_freeze(new DateTimeImmutable());
```
Comment (X-Coder264) [inline Tests/Acceptance/DoctrineAccessTokenManagerTest.php:257]: @franjo-zadelj-trikoder time freezing is not needed for revoked tokens (here and in other places too).

File: Tests/Acceptance/DoctrineAccessTokenManagerTest.php [hunk 19]
```diff
@@ -49,36 +54,149 @@ public function testClearExpired(): void
...
+                'identifier' => '7777',
+                'dateOffset' => '-1 second',
+                'revoked' => true,
+                'expired' => true,
+            ]
         ];
     }
 
-    private function buildAccessToken(string $identifier, string $modify, Client $client): AccessToken
+    private function buildTestData(Client $client, callable $successFunction): array
```
[the remark this answers is not in the document]
Comment (X-Coder264) [reply Tests/Acceptance/DoctrineAccessTokenManagerTest.php:142]: I agree. Before I could just open the test and I'd immediately see what cases were being tested (and which tokens were for example considered as expired and which weren't). Now I don't have such an overview and I have to go through the code, check what the `buildTestData` method does and how it's supposed to be used (and check what data it builds exactly) just to see which cases are being covered by the test.

File: Tests/Acceptance/DoctrineAccessTokenManagerTest.php [hunk 20]
```diff
@@ -52,16 +52,16 @@ public function testClearExpired(): void
     private function buildClearExpiredTestData(Client $client): array
     {
         $validAccessTokens = [
-            $this->buildAccessToken('1111', '+1 day', $client),
-            $this->buildAccessToken('2222', '+1 hour', $client),
-            $this->buildAccessToken('3333', '+1 second', $client),
-            $this->buildAccessToken('4444', 'now', $client),
+            $this->buildAccessToken($client, '1111', '+1 day'),
```
Comment (X-Coder264) [inline Tests/Acceptance/DoctrineAccessTokenManagerTest.php:55]: @franjo-zadelj-trikoder What is the purpose of the signature change? As far as I can see it is not needed and just adds noise to the diff.

File: Tests/Acceptance/DoctrineAccessTokenManagerTest.php [hunk 21]
```diff
@@ -109,25 +160,29 @@ public function testClearExpiredWithRefreshToken(): void
...
     }
 
     private function buildClearExpiredTestDataWithRefreshToken(Client $client): array
     {
         $validRefreshTokens = [
-            $this->buildRefreshToken('1111', '+1 day', $client),
-            $this->buildRefreshToken('2222', '+1 hour', $client),
-            $this->buildRefreshToken('3333', '+1 second', $client),
-            $this->buildRefreshToken('4444', 'now', $client),
+            $this->buildRefreshToken($client, '1111', '+1 day'),
```
Comment (X-Coder264) [inline Tests/Acceptance/DoctrineAccessTokenManagerTest.php:176]: @franjo-zadelj-trikoder What is the purpose of the signature change?

File: Tests/Acceptance/DoctrineAuthCodeManagerTest.php [hunk 22]
```diff
@@ -52,16 +52,16 @@ public function testClearExpired(): void
     private function buildClearExpiredTestData(Client $client): array
     {
         $validAuthCodes = [
-            $this->buildAuthCode('1111', '+1 day', $client),
-            $this->buildAuthCode('2222', '+1 hour', $client),
-            $this->buildAuthCode('3333', '+1 second', $client),
-            $this->buildAuthCode('4444', 'now', $client),
+            $this->buildAuthCode($client, '1111', '+1 day'),
```
Comment (X-Coder264) [inline Tests/Acceptance/DoctrineAuthCodeManagerTest.php:55]: @franjo-zadelj-trikoder What is the purpose of the signature change?

File: Tests/Acceptance/DoctrineRefreshTokenManagerTest.php [hunk 23]
```diff
@@ -55,16 +55,16 @@ public function testClearExpired(): void
     private function buildClearExpiredTestData(Client $client): array
     {
         $validRefreshTokens = [
-            $this->buildRefreshToken('1111', '+1 day', $client),
-            $this->buildRefreshToken('2222', '+1 hour', $client),
-            $this->buildRefreshToken('3333', '+1 second', $client),
-            $this->buildRefreshToken('4444', 'now', $client),
+            $this->buildRefreshToken($client, '1111', '+1 day'),
```
Comment (X-Coder264) [inline Tests/Acceptance/DoctrineRefreshTokenManagerTest.php:58]: @franjo-zadelj-trikoder What is the purpose of the signature change?

File: Tests/Unit/InMemoryAccessTokenManagerTest.php [hunk 24]
```diff
@@ -20,52 +20,128 @@ public function testClearExpired(): void
         timecop_freeze(new DateTimeImmutable());
 
         try {
-            $testData = $this->buildClearExpiredTestData();
+            $testData = $this->buildTestData(
```
Comment (X-Coder264) [inline Tests/Unit/InMemoryAccessTokenManagerTest.php:23]: @franjo-zadelj-trikoder This still uses the `buildTestData` stuff which we've agreed that should be removed, but it is still present here.

File: Tests/Unit/InMemoryAuthCodeManagerTest.php [hunk 25]
```diff
@@ -20,53 +20,128 @@ public function testClearExpired(): void
         timecop_freeze(new DateTimeImmutable());
 
         try {
-            $testData = $this->buildClearExpiredTestData();
+            $testData = $this->buildTestData(
```
Comment (X-Coder264) [inline Tests/Unit/InMemoryAuthCodeManagerTest.php:23]: @franjo-zadelj-trikoder This still uses the `buildTestData` stuff which we've agreed that should be removed, but it is still present here.

File: Tests/Unit/InMemoryAuthCodeManagerTest.php [hunk 26]
```diff
@@ -20,53 +20,128 @@ public function testClearExpired(): void
...
+            if ($successFunction($item)) {
+                $response['output'][$identifier] = $authCode;
+            }
+        }
+
+        return $response;
     }
 
-    private function buildAuthCode(string $identifier, string $modify): AuthorizationCode
+    private function buildAuthCode(string $identifier, string $modify, bool $revoked): AuthorizationCode
```
Comment (X-Coder264) [inline Tests/Unit/InMemoryAuthCodeManagerTest.php:123]: @franjo-zadelj-trikoder Another inconsistency. In some cases the $revoked argument has a default value `false`, here it doesn't. Please make that consistent.

File: Tests/Unit/InMemoryRefreshTokenManagerTest.php [hunk 27]
```diff
@@ -21,47 +21,109 @@ public function testClearExpired(): void
...
+            if ($successFunction($item)) {
+                $response['output'][$identifier] = $refreshToken;
+            }
+        }
+
+        return $response;
     }
 
-    private function buildRefreshToken(string $identifier, string $modify): RefreshToken
+    private function buildRefreshToken(string $identifier, string $modify, bool $revoked): RefreshToken
```
Comment (X-Coder264) [inline Tests/Unit/InMemoryRefreshTokenManagerTest.php:124]: @franjo-zadelj-trikoder Another inconsistency. In some cases the $revoked argument has a default value `false`, here it doesn't. Please make that consistent.

File: Tests/Unit/InMemoryRefreshTokenManagerTest.php [hunk 28]
```diff
@@ -21,47 +21,109 @@ public function testClearExpired(): void
         timecop_freeze(new DateTimeImmutable());
 
         try {
-            $testData = $this->buildClearExpiredTestData();
+            $testData = $this->buildTestData(
```
Comment (X-Coder264) [inline Tests/Unit/InMemoryRefreshTokenManagerTest.php:24]: @franjo-zadelj-trikoder This still uses the `buildTestData` stuff which we've agreed that should be removed, but it is still present here.

**Review by X-Coder264 — CHANGES_REQUESTED** [review]

File: Tests/Acceptance/DoctrineAccessTokenManagerTest.php [hunk 29]
```diff
@@ -31,7 +31,12 @@ public function testClearExpired(): void
         timecop_freeze(new DateTimeImmutable());
 
         try {
-            $testData = $this->buildClearExpiredTestData($client);
+            $testData = $this->buildTestData(
+                $client,
+                function (array $item): bool {
```
[the remark this answers is not in the document]
Comment (franjo-zadelj-trikoder) [reply Tests/Acceptance/DoctrineAccessTokenManagerTest.php]: accepted

File: Tests/Acceptance/DoctrineAccessTokenManagerTest.php [hunk 30]
```diff
@@ -109,45 +232,79 @@ public function testClearExpiredWithRefreshToken(): void
             timecop_return();
         }
 
+        $em->clear();
```
[the remark this answers is not in the document]
Comment (franjo-zadelj-trikoder) [reply Tests/Acceptance/DoctrineAccessTokenManagerTest.php]: not necessary, forgot to remove after debug

File: Tests/Unit/InMemoryAccessTokenManagerTest.php [hunk 31]
```diff
@@ -20,52 +20,134 @@ public function testClearExpired(): void
...
         );
+
+        if ($revoked) {
+            $accessToken->revoke();
+        }
+
+        return $accessToken;
+    }
+
+    private function compareOutput(array $output, InMemoryAccessTokenManager $inMemoryAccessTokenManager): void
```
[the remark this answers is not in the document]
Comment (franjo-zadelj-trikoder) [reply Tests/Unit/InMemoryAccessTokenManagerTest.php]: renamed to **assertManagerContainsExpectedData**

File: Command/ClearRevokedTokensCommand.php [hunk 32]
```diff
@@ -0,0 +1,132 @@
...
+                    'Auth codes deleted: %s.',
+                    $affected
+                )
+            );
+        }
+
+        return 0;
+    }
+
+    private function checkMethod(OutputInterface $output, object $obj, string $methodName = 'clearRevoked'): bool
```
[the remark this answers is not in the document]
Comment (franjo-zadelj-trikoder) [reply Command/ClearRevokedTokensCommand.php]: accepted

File: Tests/Unit/InMemoryRefreshTokenManagerTest.php [hunk 33]
```diff
@@ -21,47 +21,115 @@ public function testClearExpired(): void
...
 
-        $expiredRefreshTokens = [
-            '5555' => $this->buildRefreshToken('5555', '-1 day'),
-            '6666' => $this->buildRefreshToken('6666', '-1 hour'),
-            '7777' => $this->buildRefreshToken('7777', '-1 second'),
-        ];
+        $response = [];
+        foreach ($data as $item) {
+            $identifier = $item['identifier'];
+            $RefreshToken = $this->buildRefreshToken(
```
[the remark this answers is not in the document]
Comment (franjo-zadelj-trikoder) [reply Tests/Unit/InMemoryRefreshTokenManagerTest.php]: accepted

File: Tests/Unit/InMemoryAuthCodeManagerTest.php [hunk 34]
```diff
@@ -20,53 +20,134 @@ public function testClearExpired(): void
...
 
-        $expiredAuthCodes = [
-            '5555' => $this->buildAuthCode('5555', '-1 day'),
-            '6666' => $this->buildAuthCode('6666', '-1 hour'),
-            '7777' => $this->buildAuthCode('7777', '-1 second'),
-        ];
+        $response = [];
+        foreach ($data as $item) {
+            $identifier = $item['identifier'];
+            $AuthCode = $this->buildAuthCode(
```
[the remark this answers is not in the document]
Comment (franjo-zadelj-trikoder) [reply Tests/Unit/InMemoryAuthCodeManagerTest.php]: accepted

File: Tests/Acceptance/DoctrineAccessTokenManagerTest.php (hunk 19 above)
[the remark this answers is not in the document]
Comment (franjo-zadelj-trikoder) [reply Tests/Acceptance/DoctrineAccessTokenManagerTest.php]: accepted

File: Tests/Acceptance/DoctrineAccessTokenManagerTest.php (hunk 20 above)
Comment (franjo-zadelj-trikoder) [reply Tests/Acceptance/DoctrineAccessTokenManagerTest.php]: was changed in the previous version when it was completely different logic, so it was left as it was. reverted to the initial state.

File: Tests/Acceptance/DoctrineAccessTokenManagerTest.php (hunk 21 above)
Comment (franjo-zadelj-trikoder) [reply Tests/Acceptance/DoctrineAccessTokenManagerTest.php]: same as above

File: Tests/Acceptance/DoctrineAuthCodeManagerTest.php (hunk 22 above)
Comment (franjo-zadelj-trikoder) [reply Tests/Acceptance/DoctrineAuthCodeManagerTest.php]: same as above

File: Tests/Acceptance/DoctrineRefreshTokenManagerTest.php (hunk 23 above)
Comment (franjo-zadelj-trikoder) [reply Tests/Acceptance/DoctrineRefreshTokenManagerTest.php]: same as above

File: Tests/Unit/InMemoryAccessTokenManagerTest.php (hunk 24 above)
Comment (franjo-zadelj-trikoder) [reply Tests/Unit/InMemoryAccessTokenManagerTest.php]: I missed all of those in-memory test when reverting, it's ok now

File: Tests/Unit/InMemoryAuthCodeManagerTest.php (hunk 25 above)
Comment (franjo-zadelj-trikoder) [reply Tests/Unit/InMemoryAuthCodeManagerTest.php]: explained above

File: Tests/Unit/InMemoryAuthCodeManagerTest.php (hunk 26 above)
Comment (franjo-zadelj-trikoder) [reply Tests/Unit/InMemoryAuthCodeManagerTest.php]: explained above

File: Tests/Unit/InMemoryRefreshTokenManagerTest.php (hunk 27 above)
Comment (franjo-zadelj-trikoder) [reply Tests/Unit/InMemoryRefreshTokenManagerTest.php]: explained above

File: Tests/Unit/InMemoryRefreshTokenManagerTest.php (hunk 28 above)
Comment (franjo-zadelj-trikoder) [reply Tests/Unit/InMemoryRefreshTokenManagerTest.php]: explained above

File: Tests/Acceptance/ClearRevokedTokensCommandTest.php [hunk 35]
```diff
@@ -0,0 +1,184 @@
...
+            AuthorizationCode::class,
+            $this->client->getContainer()->get(AuthorizationCodeManagerInterface::class)->find(
+                FixtureFactory::FIXTURE_AUTH_CODE_REVOKED
+            )
+        );
+    }
+
+    public function testClearRevokedRefreshTokens(): void
+    {
+        $output = $this->executeCommand([
```
Comment (HypeMC) [inline Tests/Acceptance/ClearRevokedTokensCommandTest.php:106]: @franjo-zadelj-trikoder The method is called `executeCommand`, yet it clears the entity manager as well, the two seem unrelated. Perhaps it'd be better to call `clearEntityManager` explicitly after `executeCommand`.

File: Tests/Acceptance/DoctrineAccessTokenManagerTest.php [hunk 36]
```diff
@@ -109,9 +160,13 @@ public function testClearExpiredWithRefreshToken(): void
...
 
+        $mapFunction = static function (RefreshToken $refreshToken): string {
+            return $refreshToken->getIdentifier();
+        };
+
         $this->assertSame(
-            $testData['output'],
-            $em->getRepository(RefreshToken::class)->findBy(['accessToken' => null], ['identifier' => 'ASC'])
+            array_map($mapFunction, $testData['output']),
+            array_map($mapFunction, $em->getRepository(RefreshToken::class)->findBy(['accessToken' => null], ['identifier' => 'ASC']))
```
Comment (HypeMC) [inline Tests/Acceptance/DoctrineAccessTokenManagerTest.php:169]: @franjo-zadelj-trikoder Why is this change needed?

File: Tests/Acceptance/ClearRevokedTokensCommandTest.php (hunk 35 above)
Comment (franjo-zadelj-trikoder) [reply Tests/Acceptance/ClearRevokedTokensCommandTest.php:109]: @HypeMC maybe rename it to "executeCommandAndClear"? I'm trying to avoid unnecessary code duplication if possible.

File: Tests/Acceptance/DoctrineAccessTokenManagerTest.php (hunk 36 above)
Comment (franjo-zadelj-trikoder) [reply Tests/Acceptance/DoctrineAccessTokenManagerTest.php:169]: @HypeMC it's not anymore, I think in one iteration it was. Gonna revert.

File: Tests/Acceptance/ClearRevokedTokensCommandTest.php (hunk 35 above)
Comment (franjo-zadelj-trikoder) [reply Tests/Acceptance/ClearRevokedTokensCommandTest.php:111]: @HypeMC done

**Review by HypeMC — APPROVED** [review]

**Review by X-Coder264 — APPROVED** [review]

**Review by X-Coder264 — CHANGES_REQUESTED** [review]
@franjo-zadelj-trikoder The pipeline is broken:

```
dev/bin/php-test composer test -- --colors=always --coverage-clover=coverage.xml --debug

> phpunit '--colors=always' '--coverage-clover=coverage.xml' '--debug'

PHP Parse error:  syntax error, unexpected ')' in /app/src/Tests/Acceptance/ClearRevokedTokensCommandTest.php on line 187

Parse error: syntax error, unexpected ')' in /app/src/Tests/Acceptance/ClearRevokedTokensCommandTest.php on line 187

Script phpunit handling the test event returned with error code 255

The command "dev/bin/php-test composer test -- --colors=always --coverage-clover=coverage.xml --debug" exited with 255.
```

**Review by X-Coder264 — APPROVED** [review]
